validator/stateless_block_validator.go

```diff
@@ -274,37 +315,37 @@ type BatchInfo struct {
 	Data   []byte
 }
 
+func stateLogFunc(targetHeader, header *types.Header, hasState bool) {
+	if targetHeader == nil || header == nil {
+		return
+	}
+	gap := targetHeader.Number.Int64() - header.Number.Int64()
+	step := int64(500)
+	stage := "computing state"
+	if !hasState {
+		step = 3000
+		stage = "looking for full block"
+	}
+	if (gap >= step) && (gap%step == 0) {
+		log.Info("Setting up validation", "stage", stage, "current", header.Number, "target", targetHeader.Number)
+	}
+}
+
 // If msg is nil, this will record block creation up to the point where message would be accessed (for a "too far" proof)
-func RecordBlockCreation(
+func (v *StatelessBlockValidator) RecordBlockCreation(
 	ctx context.Context,
-	blockchain *core.BlockChain,
-	inboxReader InboxReaderInterface,
 	prevHeader *types.Header,
 	msg *arbstate.MessageWithMetadata,
-	producePreimages bool,
+	keepReference bool,
```

Comment:
Maybe add a comment that on error this will dereference prevHeader even if keepReference is true